### PR TITLE
Update docker_cli_build_unix_test

### PR DESCRIPTION
**Pull request by HuKeping**
part of #16756

Signed-off-by: Hu Keping hukeping@huawei.com

### CODE REVIEW

File: integration-cli/docker_cli_build_unix_test.go [hunk 1]
```diff
@@ -39,34 +37,36 @@ func (s *DockerSuite) TestBuildResourceConstraintsAreUsed(c *check.C) {
...
-		c.Fatal(err, cfg)
-	}
-	if c1.Memory != 67108864 || c1.MemorySwap != -1 || c1.CpusetCpus != "0" || c1.CpusetMems != "0" || c1.CPUShares != 100 || c1.CPUQuota != 8000 || c1.Ulimits[0].Name != "nofile" || c1.Ulimits[0].Hard != 42 {
-		c.Fatalf("resource constraints not set properly:\nMemory: %d, MemSwap: %d, CpusetCpus: %s, CpusetMems: %s, CPUShares: %d, CPUQuota: %d, Ulimits: %s",
-			c1.Memory, c1.MemorySwap, c1.CpusetCpus, c1.CpusetMems, c1.CPUShares, c1.CPUQuota, c1.Ulimits[0])
-	}
+	err = json.Unmarshal([]byte(cfg), &c1)
+	c.Assert(err, checker.IsNil, check.Commentf(cfg))
+
+	c.Assert(c1.Memory, checker.Equals, int64(64*1024*1024), check.Commentf("resource constraints not set properly for Memory"))
```
Comment (WeiZhang555) [inline integration-cli/docker_cli_build_unix_test.go]: You can put static text in comment, that will make it clearer, e.g.
```
// resource constraints not set properly for Memory
c.Assert(c1.Memory, checker.Equals, int64(64*1024*1024))
```
the comment will be printed on screen when assert failed.

File: integration-cli/docker_cli_build_unix_test.go [hunk 2]
```diff
@@ -39,34 +37,36 @@ func (s *DockerSuite) TestBuildResourceConstraintsAreUsed(c *check.C) {
...
-		c.Fatalf("resource constraints leaked from build:\nMemory: %d, MemSwap: %d, CpusetCpus: %s, CpusetMems: %s, CPUShares: %d, CPUQuota: %d, Ulimits: %s",
-			c2.Memory, c2.MemorySwap, c2.CpusetCpus, c2.CpusetMems, c2.CPUShares, c2.CPUQuota, c2.Ulimits)
+	c.Assert(err, checker.IsNil)
 
-	}
+	var c2 hostConfig
+	err = json.Unmarshal([]byte(cfg), &c2)
+	c.Assert(err, checker.IsNil, check.Commentf(cfg))
 
+	c.Assert(c2.Memory, check.Not(checker.Equals), int64(64*1024*1024), check.Commentf("resource leaked from build for Memory"))
```
Comment (WeiZhang555) [inline integration-cli/docker_cli_build_unix_test.go]: Same as above, the static text can be put in comment right above the assert.

**Comment by calavera** [comment]
LGTM. There is no need to change the assert comments, they are okay where they are right now.

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**